@here/harp-mapview/lib/Utils.ts

```diff
@@ -201,6 +193,59 @@ export namespace MapViewUtils {
         );
     }
 
+    /**
+     * Calculate target (focus) point geo-coordinates for given camera.
+     * @see getTargetPositionFromCamera
+     *
+     * @param camera The camera looking on target point.
+     * @param projection The geo-projection used.
+     * @param elevation Optional elevation above (or below) sea level measured in world units.
+     */
+    export function getTargetCoordinatesFromCamera(
+        camera: THREE.Camera,
+        projection: Projection,
+        elevation?: number
+    ): GeoCoordinates | null {
+        // This function does almost the same as:
+        // rayCastGeoCoordinates(mapView, 0, 0)
+        // but in more gentle and performance wise manner
+        const targetWorldPos = getTargetPositionFromCamera(camera, projection, elevation);
+        if (targetWorldPos !== null) {
+            return projection.unprojectPoint(targetWorldPos);
+        }
+        return null;
+    }
+
+    /**
+     * Calculate target (focus) point world coordinates for given camera position and orientation.
+     * @param camera The camera looking on target point.
+     * @param projection The geo-projection used.
+     * @param elevation Optional elevation above (or below) sea level in world units.
+     */
+    export function getTargetPositionFromCamera(
+        camera: THREE.Camera,
+        projection: Projection,
+        elevation?: number
+    ): THREE.Vector3 | null {
+        const cameraPos = cache.vector3[0].copy(camera.position);
+        const cameraLookAt = camera.getWorldDirection(cache.vector3[1]);
+        rayCaster.set(cameraPos, cameraLookAt);
+        if (elevation !== undefined) {
+            groundPlane.constant = -elevation;
+            groundSphere.radius += elevation;
+        }
+        const targetWorldPos = new THREE.Vector3();
+        const result =
+            projection.type === ProjectionType.Planar
```

Comment:
Maybe check my PR: https://github.com/heremaps/harp.gl/pull/1178/files#diff-ab66d6ee9a32de1a17a5622106fd3b08R2432-R2522

I compute and cache the targetWorld in MapView.updateCameras. I think this you should also do since the target point is used in many places